gobblin-api/src/main/java/org/apache/gobblin/password/PasswordManager.java

```diff
@@ -160,27 +204,36 @@ public String encryptPassword(String plain) {
   }
 
   /**
-   * Decrypt an encrypted password. A master password must have been provided in the constructor.
+   * Decrypt an encrypted password. A master password file must have been provided in the constructor.
    * @param encrypted An encrypted password.
    * @return The decrypted password.
    */
   public String decryptPassword(String encrypted) {
-    Preconditions.checkArgument(this.encryptor.isPresent(),
+    Preconditions.checkArgument(this.masterPasswords.size() > 0,
         "A master password needs to be provided for decrypting passwords.");
 
-    try {
-      return this.encryptor.get().decrypt(encrypted);
-    } catch (Exception e) {
-      throw new RuntimeException("Failed to decrypt password " + encrypted, e);
+    for (String masterPassword : masterPasswords) {
+      try (Closer closer = Closer.create()) {
+        this.encryptor = useStrongEncryptor ? Optional.of((TextEncryptor) new StrongTextEncryptor())
```

Comment:
These should be cached like in the old code. Either construct them in the constructor (and loop through the encryptors list instead of masterPasswords) or cache them in a map here. The number of passwords should be small, so I think creating the encryptors at construction time will be simpler than maintaining a cache.

Reply:
done